corehq/apps/registration/forms.py

```diff
@@ -632,9 +644,20 @@ def clean(self):
             raise forms.ValidationError(error)
         return cleaned_data
 
-    def _initialize_tableau_fields(self, data, domain):
+    def _initialize_tableau_fields(self, data, domain, invitation=None):
         self.tableau_form = BaseTableauUserForm(data, domain=domain)
         self.fields['tableau_group_indices'] = self.tableau_form.fields["groups"]
         self.fields['tableau_group_indices'].label = _('Tableau Groups')
         self.fields['tableau_role'] = self.tableau_form.fields['role']
         self.fields['tableau_role'].label = _('Tableau Role')
+        if invitation:
+            initial_groups_indicies = []
+            for group_index in invitation.tableau_group_ids:
```

Comment:
Nice - but I think tableau_group_ids can be `None` in which case this would error